tests/test_core.py

```diff
@@ -57,12 +59,78 @@ class test_Variable(unittest.TestCase):
 
     def test_init(self):
         """Test construction with different bounds."""
-        raise NotImplementedError()
+        v = relentless.core.Variable(value=1.0, const=True)
+        self.assertAlmostEqual(v.value, 1.0)
+        self.assertEqual(v.const, True)
+
+        v = relentless.core.Variable(value=1.2, low=0.0, high=2.0)
+        self.assertAlmostEqual(v.value, 1.2)
+        self.assertAlmostEqual(v.low, 0.0)
+        self.assertAlmostEqual(v.high, 2.0)
+
+        v = relentless.core.Variable(value=-1.2, low=0.5)
+        self.assertAlmostEqual(v.value, 0.5)
+        self.assertAlmostEqual(v.low, 0.5)
```

Comment:
Could you also test the `isfree`, `atlow`, and `athigh` methods on all these variables?